### DISCUSSION TITLE
How to filter by URI path suffix?

### DISCUSSION THREAD

**Discussion by evilpan** [mitmproxy/mitmproxy#6446] [Q&A] [answered]
Hello,

There's view_filter `~u` that can match the whole URL, and `~d` match the domain, but how can I match by URI path? For example, I want to filter out flows by request suffix `.js`, `.css`, `.png`, etc. (I known `~a` can match `assets` but not very extensible). The filter I set is:

```
set view_filter '~u "\.(js|css|png)$"'
```

However sometimes the URL may contains query params, for example, `https://example.com/foo.png?size=20`. So I try to update the filter to:

```
set view_filter '~u "\.(js|css|png)(\?.*)?$"'
```

But it shows:
```
error: [17:10:51.515] Invalid filter expression: '~u "\\.(js|css|png)(\\?.*)?$"'
```

It's a valid regex and I've tried it on Python console. So, what's wrong with my filter, and what's the right way to filter by URI path suffix then?

**Comment by mhils** [comment, marked as answer]
I suppose you are only running into some escaping issues here. 
```
set view_filter '~u "\\.(js|css|png)(\\\\?.*)?$"' 
```
works as you intended. I suppose the problem is that there's a wild mix of 1) command escaping, 2) Python escaping, and 3) Regex escaping. 🙈

**Reply by evilpan** [reply]
Thank you, It works now! Though I still don't understand why the first `.` only need to be escape once but the `?` need to be escape twice. 🤦🏻‍♀️
